server/resources/resource_management.go

```diff
@@ -72,30 +72,25 @@ type ExtensionError struct {
 }
 
 //InTransaction executes function in the db transaction and set it to the context
-func InTransaction(ctx middleware.Context, dataStore db.DB, level transaction.Type, f func() error) error {
+func InTransaction(ctx middleware.Context, dataStore db.DB, level transaction.Type, fn func() error) error {
 	if ctx["transaction"] != nil {
```

Comment:
if there's a restart of event handling, make a fresh, deep-copy of the context (extensions are allowed to save some state inside)